frontend/src/components/interaction-manager/content/wallet-selection.tsx

```diff
@@ -70,55 +78,70 @@ export const WalletSelection = ({
   }
 
   return (
-    <Dialog open={true} size='lg'>
-      <div
+    <Dialog open={true} size='lg' title='Approve connection'>
```

Comment:
we don't have a translation layer in this repo. this will be taken care of later on, when the priorities allow, definitely not in this PR. 2k line changes i believe are enough.